src/libcollections/binary_heap.rs

```diff
@@ -688,6 +688,22 @@ impl<T: Ord> BinaryHeap<T> {
         }
     }
 
+    fn sift_up_ind(&mut self, start: usize, pos: usize) -> usize {
```

Comment:
You can simply adjust the `sift_up` to return the index, since its a private implementation detail and will also avoid the code duplication.